<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CaptureOptionsDialog</class>
 <widget class="QDialog" name="CaptureOptionsDialog">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>478</width>
    <height>610</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Capture Options</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="verticalSpacing">
    <number>9</number>
   </property>
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>12</number>
     </property>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>General</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QCheckBox" name="samplingCheckBox">
          <property name="text">
           <string>Enable callstack sampling</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="samplingPeriodMsHorizontalLayout">
          <item>
           <widget class="QLabel" name="samplingPeriodMsLabel">
            <property name="text">
             <string>Callstack sampling period (ms):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="samplingPeriodMsDoubleSpinBox">
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>10.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="samplingPeriodMsHorizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QWidget" name="unwindingMethodWidget" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="unwindingMethodLabel">
             <property name="text">
              <string>Callstack unwinding method:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="unwindingMethodComboBox"/>
           </item>
           <item>
            <spacer name="unwindingMethodHorizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Expanding</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>112</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="schedulerCheckBox">
          <property name="text">
           <string>Collect scheduler information</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="threadStateCheckBox">
          <property name="text">
           <string>Collect thread states</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="gpuSubmissionsCheckBox">
          <property name="text">
           <string>Trace GPU queue submissions</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="apiCheckBox">
          <property name="text">
           <string>Enable Orbit Api in target</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="introspectionCheckBox">
          <property name="text">
           <string>Enable introspection</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="dynamicInstrumentationMethodWidget" native="true">
          <property name="toolTip">
           <string>Method 'Orbit' has lower overhead but is still experimental.</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Dynamic Instrumentation method:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="dynamicInstrumentationMethodComboBox">
             <property name="accessibleName">
              <string>DynamicInstrumentationMethodComboBox</string>
             </property>
             <property name="minimumSize">
              <size>
               <width>170</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>Memory tracing</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QCheckBox" name="collectMemoryInfoCheckBox">
          <property name="text">
           <string>Collect memory usage and page faults information</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="memoryHorizontalLayout">
          <item>
           <widget class="QLabel" name="memorySamplingPeriodMsLabel">
            <property name="text">
             <string>Sampling period (ms):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="memorySamplingPeriodMsLineEdit">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="accessibleName">
             <string>MemorySamplingPeriodEdit</string>
            </property>
            <property name="inputMethodHints">
             <set>Qt::ImhDigitsOnly</set>
            </property>
            <property name="text">
             <string>10</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>60</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="memoryWarningThresholdKbLabel">
            <property name="text">
             <string>Warning threshold (kB):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="memoryWarningThresholdKbLineEdit">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>The default value is 8388608 KB, i.e., 8GB.</string>
            </property>
            <property name="inputMethodHints">
             <set>Qt::ImhDigitsOnly</set>
            </property>
            <property name="text">
             <string>8388608</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="toolTip">
        <string>This requires Orbit's Vulkan layer to be loaded by the target process.</string>
       </property>
       <property name="title">
        <string>Vulkan layer (requires Orbit's Vulkan layer to be loaded by the target)</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="localMarkerDepthHorizontalLayout">
          <item>
           <widget class="QCheckBox" name="localMarkerDepthCheckBox">
            <property name="text">
             <string>Limit local depth of markers per command buffer to:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="localMarkerDepthLineEdit">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Limits the maximum depth of Vulkan debug markers per command buffer to the given value.
Setting the value to &quot;0&quot; will disable Vulkan debug marker tracking.</string>
            </property>
            <property name="inputMethodHints">
             <set>Qt::ImhDigitsOnly</set>
            </property>
            <property name="text">
             <string>3</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="2" column="0">
    <widget class="Line" name="line">
     <property name="styleSheet">
      <string notr="true">background: #353535</string>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>These settings will only apply starting from your next capture.</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>localMarkerDepthCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>localMarkerDepthLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>390</x>
     <y>133</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>localMarkerDepthLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>CaptureOptionsDialog</receiver>
   <slot>ResetLocalMarkerDepthLineEdit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>390</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>129</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>collectMemoryInfoCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>memorySamplingPeriodMsLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>173</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>collectMemoryInfoCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>memoryWarningThresholdKbLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>413</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>memorySamplingPeriodMsLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>CaptureOptionsDialog</receiver>
   <slot>ResetMemorySamplingPeriodMsLineEditWhenEmpty()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>memoryWarningThresholdKbLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>CaptureOptionsDialog</receiver>
   <slot>ResetMemoryWarningThresholdKbLineEditWhenEmpty()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>413</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>samplingCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>samplingPeriodMsDoubleSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>80</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>samplingCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>unwindingMethodComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>110</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>ResetLocalMarkerDepthLineEdit()</slot>
  <slot>ResetMemorySamplingPeriodMsLineEditWhenEmpty()</slot>
  <slot>ResetMemoryWarningThresholdKbLineEditWhenEmpty()</slot>
 </slots>
</ui>
